### PR TITLE
Factor out some common object mocking in tests

### PR DESCRIPTION
**Pull request by gzhao9**
Hi there,

This PR is to address [issue #14768](https://github.com/spring-projects/spring-security/issues/14768), which involves reducing repeated mock object creation in tests. 

I mistakenly closed PR #15256 and then discovered that I do not have permission to reopen it, receiving the message `this branch was force pushed or recreated.`  I'm not sure if I should reopen the original or create this new PR 

Although my issue submission mentioned 4 separate draft PRs, I combined them into this PR, since they all address the same problem.

### CODE REVIEW

**Comment by gzhao9** [comment]
@jesperronn Please feel free to let me know if there is anything I need to change. I look forward to your suggestions.

**Review by jzheaux — CHANGES_REQUESTED** [review]
Thanks for the idea, @gzhao9. I think there is already a `MockSecurityContextHolderStrategy` available in `spring-security-core`. Can you please enhance that instead?

I think it would be reasonable to add the default constructor and one that takes a `SecurityContext` to that class.

Please also run `./gradlew format && ./gradlew check` before committing next. It appears there are some formatting issues that those two tasks will help you work through.

When you are done, will you please change your commits to be more descriptive in the following way. For each commit, describe what you've done and include the issue number like so:

```
Use MockSecurityContextHolderStrategy

Issue gh-14768
```

For the final commit, please use `Closes gh-14768` instead as this integrates with our build system to close the issue automatically.

File: ldap/src/test/java/org/springframework/security/ldap/authentication/ad/ActiveDirectoryLdapAuthenticationProviderTests.java [hunk 1]
```diff
@@ -90,8 +91,6 @@ public void successfulAuthenticationProducesExpectedAuthorities() throws Excepti
 	@Test
 	public void customSearchFilterIsUsedForSuccessfulAuthentication() throws Exception {
 		String customSearchFilter = "(&(objectClass=user)(sAMAccountName={0}))";
-		DirContext ctx = mock(DirContext.class);
-		given(ctx.getNameInNamespace()).willReturn("");
```
Comment (jzheaux) [inline ldap/src/test/java/org/springframework/security/ldap/authentication/ad/ActiveDirectoryLdapAuthenticationProviderTests.java:94]: I'd like the tests to remain as similar as possible behaviorally. Will you then please add this to the setup method:

```java
given(ctx.getNameInNamespace()).willReturn("");
```

File: web/src/test/java/org/springframework/security/web/access/ExceptionTranslationFilterTests.java [hunk 2]
```diff
@@ -304,7 +291,11 @@ public void setMessageSourceWhenNotNullThenCanGet() {
 		filter.messages.getMessage(code);
 		verify(source).getMessage(eq(code), any(), any());
 	}
-
+	private FilterChain mockFilterChainWiehException(Object exception) throws ServletException, IOException {
```
Comment (jzheaux) [inline web/src/test/java/org/springframework/security/web/access/ExceptionTranslationFilterTests.java:294]: I believe there is a typo here and the method should be called `mockFilterChainWithException`.

Comment (jzheaux) [inline web/src/test/java/org/springframework/security/web/access/ExceptionTranslationFilterTests.java:294]: I think this method can take `Exception` for the parameter type. Is there a reason it needs to be `Object`?

File: web/src/test/java/org/springframework/security/web/access/ExceptionTranslationFilterTests.java [hunk 3]
```diff
@@ -304,7 +291,11 @@ public void setMessageSourceWhenNotNullThenCanGet() {
...
+		FilterChain fc = mock(FilterChain.class);
+		willThrow((Throwable) exception).given(fc).doFilter(any(HttpServletRequest.class), any(HttpServletResponse.class));
```
Comment (jzheaux) [inline web/src/test/java/org/springframework/security/web/access/ExceptionTranslationFilterTests.java:296]: If you can change the parameter type, this cast should be removed.

**Review by jzheaux — CHANGES_REQUESTED** [review]
Thank you for the PR, @gzhao9. I've left feedback inline.

Additionally, will you please consider whether you can follow the same strategy for all four files when it comes to setting up mocks? It seems that in one you have a separate class, in another you have a field and a `setup` method, and in others you use a local method. It would be nice if the code can be more consistent.

Finally, will you please ensure that your commits are in the following format:

```
Factor out common SecurityContextHolderStrategy mocking
```

or similar, depending on the work done for each file.
```

**Comment by gzhao9** [comment]
> Thanks for the cleanup, @gzhao9. I think there is already a `MockSecurityContextHolderStrategy` available in `spring-security-core`. Can you please enhance that instead?
> 
> I think it would be reasonable to add the default constructor and one that takes a `SecurityContext` to that class.
> 
> Please also run `./gradlew format && ./gradlew check` before committing next. It appears there are some formatting issues that those two tasks will help you work through.
> 
> When you are done, will you please change your commits to be more descriptive in the following way. For each commit, describe what you've done and include the issue number like so:
> 
> ```
> Use MockSecurityContextHolderStrategy
> 
> Issue gh-14768
> ```
> 
> For the final commit, please use `Closes gh-14768` instead as this integrates with our build system to close the issue automatically.

I found `MockSecurityContextHolderStrategy` in `spring-security-core`. However, since they are not in the same package, I would need to change the dependency if I want to reference `MockSecurityContextHolderStrategy` in `org.springframework.security.authorization.method`. I believe adding new dependencies should be considered carefully. Therefore, I only added a private method in each test class.

**Comment by gzhao9** [comment]
I first focus on each file code modification. I will rebase commits when there is no need to modify the code or format.

**Comment by jzheaux** [comment]
Thanks for the updates, @gzhao9! I think there are two other items from the review that haven't been addressed yet. Are you able to do those as well?

**Comment by gzhao9** [comment]
> Thanks for the updates, @gzhao9! I think there are two other items from the review that haven't been addressed yet. Are you able to do those as well?

I have modified the code as required.

**Review by jzheaux — APPROVED** [review]

**Comment by jzheaux** [comment]
Thanks, @gzhao9! This is now merged into `main`.

**Pull request merged**